### ISSUE TITLE
translate-c broken for SDL

### ISSUE THREAD

**Issue by nmichaels** [ziglang/zig#5560]
Somewhere between 
93384f7428c5186a1b44c6f5f8891ce40bc46e72 and 1a90a5e63adfd585836aee2658275c02856a4427 (inclusive), translate-c started throwing an error for this minimal example:

```
const std = @import("std");
usingnamespace @cImport({
    @cInclude("SDL2/SDL.h");
});

pub fn main() anyerror!void {
    return;
}
```

built with
```
const Builder = @import("std").build.Builder;

pub fn build(b: *Builder) void {
    const mode = b.standardReleaseOptions();
    const exe = b.addExecutable("min", "main.zig");
    exe.setBuildMode(mode);
    exe.linkSystemLibrary("c");
    exe.linkSystemLibrary("SDL2");
    exe.install();
}
```

**Comment by Vexu** [comment]
I think this is a duplicate of #5523, can you try updating your Zig?

**Comment by nmichaels** [comment]
Nope, still there. Here's the actual error text:
```
$ zig build
Code Generation [524/1631] std.hash_map.HashMap([]const u8,std.build.UserInputOpindex out of bounds
/home/me/zig/zig/src-self-hosted/translate_c.zig:2669:23: 0x5577c1ca2aad in translate_c.transCase (zigstage2)
    switch_scope.cases[switch_scope.case_index] = &switch_prong.base;
                      ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:1188:43: 0x5577c1c37838 in translate_c.transStmt (zigstage2)
        .CaseStmtClass => return transCase(rp, scope, @ptrCast(*const ZigClangCaseStmt, stmt)),
                                          ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:1411:37: 0x5577c1bf4a8f in translate_c.transCompoundStmtInline (zigstage2)
        const result = try transStmt(rp, parent_scope, it[0], .unused, .r_value);
                                    ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:1419:32: 0x5577c1c99661 in translate_c.transCompoundStmt (zigstage2)
    try transCompoundStmtInline(rp, &block_scope.base, stmt, &block_scope);
                               ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:1148:55: 0x5577c1c371cf in translate_c.transStmt (zigstage2)
        .CompoundStmtClass => return transCompoundStmt(rp, scope, @ptrCast(*const ZigClangCompoundStmt, stmt)),
                                                      ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:2470:35: 0x5577c1c9d144 in translate_c.transDoWhileLoop (zigstage2)
        const node = try transStmt(rp, &loop_scope, ZigClangDoStmt_getBody(stmt), .unused, .r_value);
                                  ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:1171:48: 0x5577c1c37640 in translate_c.transStmt (zigstage2)
        .DoStmtClass => return transDoWhileLoop(rp, scope, @ptrCast(*const ZigClangDoStmt, stmt)),
                                               ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:2685:21: 0x5577c1ca2991 in translate_c.transCase (zigstage2)
    return transStmt(rp, scope, ZigClangCaseStmt_getSubStmt(stmt), .unused, .r_value);
                    ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:1188:43: 0x5577c1c37838 in translate_c.transStmt (zigstage2)
        .CaseStmtClass => return transCase(rp, scope, @ptrCast(*const ZigClangCaseStmt, stmt)),
                                          ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:1411:37: 0x5577c1bf4a8f in translate_c.transCompoundStmtInline (zigstage2)
        const result = try transStmt(rp, parent_scope, it[0], .unused, .r_value);
                                    ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:1419:32: 0x5577c1c99661 in translate_c.transCompoundStmt (zigstage2)
    try transCompoundStmtInline(rp, &block_scope.base, stmt, &block_scope);
                               ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:1148:55: 0x5577c1c371cf in translate_c.transStmt (zigstage2)
.CompoundStmtClass => return transCompoundStmt(rp, scope, @ptrCast(*const ZigClangCompoundStmt, stmt)),
                                                      ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:2610:31: 0x5577c1ca13c5 in translate_c.transSwitch (zigstage2)
    const last = try transStmt(rp, &block_scope.base, ZigClangSwitchStmt_getBody(stmt), .unused, .r_value);
                              ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:1187:47: 0x5577c1c377e9 in translate_c.transStmt (zigstage2)
        .SwitchStmtClass => return transSwitch(rp, scope, @ptrCast(*const ZigClangSwitchStmt, stmt)),
                                              ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:1411:37: 0x5577c1bf4a8f in translate_c.transCompoundStmtInline (zigstage2)
        const result = try transStmt(rp, parent_scope, it[0], .unused, .r_value);
                                    ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:618:28: 0x5577c1b99ea6 in translate_c.visitFnDecl (zigstage2)
    transCompoundStmtInline(rp, &block_scope.base, casted_body, &block_scope) catch |err| switch (err) {
                           ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:482:31: 0x5577c1b2caf6 in translate_c.declVisitor (zigstage2)
            return visitFnDecl(c, @ptrCast(*const ZigClangFunctionDecl, decl));
                              ^
/home/me/zig/zig/src-self-hosted/translate_c.zig:465:16: 0x5577c1b27e20 in translate_c.declVisitorC (zigstage2)
    declVisitor(c, decl) catch |err| {
               ^
min...The following command terminated unexpectedly:
/home/me/bin/zig build-exe /home/me/zig/minsdl/main.zig --library c --library SDL2 --cache-dir /home/me/zig/minsdl/zig-cache --name min -I /usr/include/SDL2 -D _REENTRANT --cache on 

Build failed. The following command failed:
/home/me/zig/minsdl/zig-cache/o/ooG4rSNgTdNT6AFV_6KO1RkATGs0YTMPAExZyGmTn2AV6QcjgLgIjq2S3KHaRG88/build /home/me/bin/zig /home/me/zig/minsdl /home/me/zig/minsdl/zig-cache
```

This is libsdl2-dev version 2.0.12+dfsg1-1 on Debian, if it matters.

**Comment by Vexu** [comment]
What does `zig version` print?

**Comment by nmichaels** [comment]
0.6.0, but I just built b3980bdcc89337ccd08b3e275b596ebf22d8cc9c
When I use the actual 0.6.0 tag, it works.

**Comment by Vexu** [comment]
That's too old, the fix was merged three days ago: 499df9680ceea203602ba1aebc475b047714dbf4.

**Comment by nmichaels** [comment]
Huh, weird. I had just pulled master, but I guess my repo was messed up somehow. A fresh clone and rebuild confirmed that you're right.
